backend/src/main/java/ai/verta/modeldb/experiment/subtypes/CreateExperimentHandler.java

```diff
@@ -0,0 +1,291 @@
+import ai.verta.modeldb.experimentRun.subtypes.CodeVersionHandler;
+import ai.verta.modeldb.experimentRun.subtypes.TagsHandler;
+import ai.verta.modeldb.metadata.MetadataServiceImpl;
+import ai.verta.modeldb.utils.HandlerUtil;
+import ai.verta.modeldb.utils.ModelDBUtils;
+import ai.verta.uac.Entities;
+import ai.verta.uac.ResourceType;
+import ai.verta.uac.Resources;
+import ai.verta.uac.RoleBinding;
+import ai.verta.uac.RoleScope;
+import ai.verta.uac.ServiceEnum;
+import ai.verta.uac.SetRoleBinding;
+import ai.verta.uac.UserInfo;
+import java.util.Calendar;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.transaction.TransactionIsolationLevel;
+
+public class CreateExperimentHandler extends HandlerUtil {
+
+  private static Logger LOGGER = LogManager.getLogger(CreateExperimentHandler.class);
+
+  private final Executor executor;
+  private final FutureJdbi jdbi;
+  private final Config config;
+  private final UAC uac;
+
+  private final AttributeHandler attributeHandler;
+  private final TagsHandler tagsHandler;
+  private final ArtifactHandler artifactHandler;
+  private final CodeVersionHandler codeVersionHandler;
+
+  public CreateExperimentHandler(
+      Executor executor,
+      FutureJdbi jdbi,
+      Config config,
+      UAC uac,
+      AttributeHandler attributeHandler,
+      TagsHandler tagsHandler,
+      ArtifactHandler artifactHandler) {
+    this.executor = executor;
+    this.jdbi = jdbi;
+    this.config = config;
+    this.uac = uac;
+
+    this.attributeHandler = attributeHandler;
+    this.tagsHandler = tagsHandler;
+    this.artifactHandler = artifactHandler;
+    this.codeVersionHandler = new CodeVersionHandler(executor, jdbi, "experiment");
+  }
+
+  public InternalFuture<Experiment> convertCreateRequest(
+      final CreateExperiment request, UserInfo userInfo) {
+    return InternalFuture.completedInternalFuture(getExperimentFromRequest(request, userInfo));
+  }
+
+  /**
+   * Method to convert createExperiment request to Experiment object. This method generates the
+   * experiment Id using UUID and puts it in Experiment object.
+   *
+   * @param request : CreateExperiment
+   * @return Experiment
+   */
+  private Experiment getExperimentFromRequest(CreateExperiment request, UserInfo userInfo) {
+
+    String errorMessage = null;
+    if (request.getProjectId().isEmpty()) {
+      errorMessage = "Project ID not found in CreateExperiment request";
+    } else if (request.getName().isEmpty()) {
+      request = request.toBuilder().setName(MetadataServiceImpl.createRandomName()).build();
+    }
+
+    if (errorMessage != null) {
+      throw new InvalidArgumentException(errorMessage);
+    }
```

Comment:
This code as written works but I'd rather us do all the fatal checks in one block and non-fatal checks in a separate block.

```suggestion
    if (request.getProjectId().isEmpty()) {
      errorMessage = "Project ID not found in CreateExperiment request";
    }

    if (errorMessage != null) {
      throw new InvalidArgumentException(errorMessage);
    }
    
    if (request.getName().isEmpty()) {
      request = request.toBuilder().setName(MetadataServiceImpl.createRandomName()).build();
    }
```

Reply:
Fixed